backend/controllers/container.rb

```diff
@@ -138,4 +138,23 @@ class ArchivesSpaceService < Sinatra::Base
     end
   end
 
+
+  Endpoint.post('/repositories/:repo_id/top_containers/bulk/labels')
+  .description("Bulk label data")
+  .params(["record_uris", [String], "A list of container uris"],
+          ["repo_id", :repo_id])
+  .permissions([])
+  .returns([200, "Container data for label printing"]) \
+  do
+    begin
+      results = params[:record_uris].map do |uri|
+        topcon = resolve_references(TopContainer.to_jsonmodel(JSONModel(:top_container).id_for(uri)), 'container_locations')
+      end
+
```

Comment:
It'd be nice to batch this where possible.  I tried something like this and I think it worked:

      ids = params[:record_uris].map {|uri| JSONModel(:top_container).id_for(uri)}
      top_containers = TopContainer.sequel_to_jsonmodel(TopContainer.filter(:id => ids).all).map{|tc| tc.to_hash(:trusted)}
      resolved = resolve_references(top_containers, ['container_locations', 'linked_records'])
      json_response(:results => resolved)

Usually this is all hidden within the JSONModel.  Maybe creating a JSONModel to act as a the report might help? 

I've also tried adding 'linked_records' records to the list to resolve.  If this was a 'ref' in the schema then it could be batch retrieved also.